test_pct_max_many_rows fails on intel 32bit with memory error

Pytest output from the two failing tests:
```
[ 3680s] =================================== FAILURES ===================================
[ 3680s] _______________________ TestRank.test_pct_max_many_rows ________________________
[ 3680s] 
[ 3680s] self = <pandas.tests.frame.test_rank.TestRank object at 0x54f6d7ec>
[ 3680s] 
[ 3680s]     @pytest.mark.single
[ 3680s]     def test_pct_max_many_rows(self):
[ 3680s]         # GH 18271
[ 3680s]         df = DataFrame({'A': np.arange(2**24 + 1),
[ 3680s]                         'B': np.arange(2**24 + 1, 0, -1)})
[ 3680s] >       result = df.rank(pct=True).max()
[ 3680s] 
[ 3680s] ../../BUILDROOT/python-pandas-0.24.1-13.1.i386/usr/lib/python2.7/site-packages/pandas/tests/frame/test_rank.py:317: 
[ 3680s] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[ 3680s] ../../BUILDROOT/python-pandas-0.24.1-13.1.i386/usr/lib/python2.7/site-packages/pandas/core/generic.py:8335: in rank
[ 3680s]     return ranker(self)
[ 3680s] ../../BUILDROOT/python-pandas-0.24.1-13.1.i386/usr/lib/python2.7/site-packages/pandas/core/generic.py:8327: in ranker
[ 3680s]     pct=pct)
[ 3680s] ../../BUILDROOT/python-pandas-0.24.1-13.1.i386/usr/lib/python2.7/site-packages/pandas/core/algorithms.py:861: in rank
[ 3680s]     ascending=ascending, na_option=na_option, pct=pct)
[ 3680s] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[ 3680s] 
[ 3680s] >   ranks = np.empty((n, k), dtype='f8')
[ 3680s] E   MemoryError
[ 3680s] 
[ 3680s] pandas/_libs/algos_rank_helper.pxi:806: MemoryError
[ 3680s] ____________________________ test_pct_max_many_rows ____________________________
[ 3680s] 
[ 3680s]     @pytest.mark.single
[ 3680s]     def test_pct_max_many_rows():
[ 3680s]             # GH 18271
[ 3680s]             s = Series(np.arange(2**24 + 1))
[ 3680s] >           result = s.rank(pct=True).max()
[ 3680s] 
[ 3680s] ../../BUILDROOT/python-pandas-0.24.1-13.1.i386/usr/lib/python2.7/site-packages/pandas/tests/series/test_rank.py:505: 
[ 3680s] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[ 3680s] ../../BUILDROOT/python-pandas-0.24.1-13.1.i386/usr/lib/python2.7/site-packages/pandas/core/generic.py:8335: in rank
[ 3680s]     return ranker(self)
[ 3680s] ../../BUILDROOT/python-pandas-0.24.1-13.1.i386/usr/lib/python2.7/site-packages/pandas/core/generic.py:8327: in ranker
[ 3680s]     pct=pct)
[ 3680s] ../../BUILDROOT/python-pandas-0.24.1-13.1.i386/usr/lib/python2.7/site-packages/pandas/core/algorithms.py:857: in rank
[ 3680s]     na_option=na_option, pct=pct)
[ 3680s] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[ 3680s] 
[ 3680s] >   sorted_data = values.take(_as)
[ 3680s] E   MemoryError
[ 3680s] 
[ 3680s] pandas/_libs/algos_rank_helper.pxi:712: MemoryError
[ 3681s] =============================== warnings summary ===============================
```

Full build log: 
[pandas.txt](https://github.com/pandas-dev/pandas/files/2884130/pandas.txt)

--- a/pandas/tests/frame/test_rank.py
+++ b/pandas/tests/frame/test_rank.py
@@ -310,6 +310,7 @@ class TestRank(TestData):
         tm.assert_frame_equal(result, expected)
 
     @pytest.mark.single
+    @pytest.mark.high_memory
     def test_pct_max_many_rows(self):
         # GH 18271
         df = DataFrame({'A': np.arange(2**24 + 1),

--- a/pandas/tests/series/test_rank.py
+++ b/pandas/tests/series/test_rank.py
@@ -499,6 +499,7 @@ def test_rank_first_pct(dtype, ser, exp):
 
 
 @pytest.mark.single
+@pytest.mark.high_memory
 def test_pct_max_many_rows():
         # GH 18271
         s = Series(np.arange(2**24 + 1))

--- a/pandas/tests/test_algos.py
+++ b/pandas/tests/test_algos.py
@@ -1484,6 +1484,7 @@ class TestRank(object):
             algos.rank(arr)
 
     @pytest.mark.single
+    @pytest.mark.high_memory
     @pytest.mark.parametrize('values', [
         np.arange(2**24 + 1),
         np.arange(2**25 + 2).reshape(2**24 + 1, 2)],

